Default to SSL verify on

leoetlino opened the pull request:
Defaulting to SSL verification off, *and* forcing it to be off even when the emulated software asks us to enable it is very bad behaviour, inaccurate and insecure.

JMC47 commented:
I need this so I can put my credit card info into Dolphin to buy Wii points.

sepalani commented on Source/Core/Core/IOS/Network/SSL.cpp:
```diff
@@ -211,7 +211,7 @@ IPCCommandResult NetSSL::IOCtlV(const IOCtlVRequest& request)
       mbedtls_ssl_set_session(&ssl->ctx, &ssl->session);
 
-      if (SConfig::GetInstance().m_SSLVerifyCert && verifyOption)
+      if (SConfig::GetInstance().m_SSLVerifyCert || verifyOption)
```
LBTM. If the console requires the certificate not to be verified, it will verify it if ```m_SSLVerifyCert``` is set.

The purpose of that option is to force that insecure behavior when it's set to false (useful for debugging, homebrew, etc.).

JosJuice commented:
What's unfortunate is that setting the setting to true by default won't affect the settings of anyone who has run any Dolphin version where it's false by default (assuming they don't change the value manually, which most people don't).

leoetlino commented:
Reverted to the previous behaviour (config option && requested by emulated software).

I have also updated the option name to reset to the default option; otherwise we cannot ensure our users are safe with SSL.

sepalani commented on Source/Core/Core/ConfigManager.cpp:
```diff
@@ -325,7 +325,7 @@ void SConfig::SaveNetworkSettings(IniFile& ini)
   network->Set("SSLDumpRead", m_SSLDumpRead);
   network->Set("SSLDumpWrite", m_SSLDumpWrite);
-  network->Set("SSLVerifyCert", m_SSLVerifyCert);
+  network->Set("SSLVerifyCertificates", m_SSLVerifyCert);
```
To keep it consistent, shouldn't the other options be renamed as well? ```SSLDumpRootCertificateAuthorities``` and ```SSLDumpPeerCertificates```? Same concerns for the member variables.

leoetlino replied:
The only purpose of renaming it is to reset the default. I don't think we should update the other options just for consistency

I wouldn't update the member variables either, because doing so will force everyone to recompile half of Dolphin.

sepalani replied:
TBH, I think I was probably one of the few using it and if no body complained about that lack of SSL verifications since the last 2 years. It seems perfectly safe just to change the default value.

Knowing that most lambda users don't care about that feature and I don't think that the cleanest way to reset the default value. In other words, the "Network" entry shouldn't be present in their configuration file. All my portable builds of Dolphin after that addition don't have it so unless that behavior was added recently it shouldn't be the case.

JosJuice replied:
We call SaveNetworkSettings every time any settings are going to be saved to Dolphin.ini, so I don't know how you've managed to avoid getting a Network section. (Or am I misinterpreting your comment?) I have a Network section with `SSLVerifyCert = false` even though I've never touched it.

sepalani replied:
That's strange because my portable builds aren't affected (and these versions have this feature, so that's strange). LGTM, then.
